data/scripts.js

```diff
@@ -296,165 +259,259 @@ let BattleScripts = {
 			this.singleEvent('AfterMoveSecondarySelf', move, null, pokemon, target, move);
 			this.runEvent('AfterMoveSecondarySelf', pokemon, target, move);
 		}
+
 		return true;
 	},
-	tryMoveHit(target, pokemon, move) {
-		this.setActiveMove(move, pokemon, target);
+	trySpreadMoveHit(targets, pokemon, move) {
+		if (!targets.length) {
+			this.attrLastMove('[notarget]');
+			this.add(this.gen >= 5 ? '-fail' : '-notarget', pokemon);
+			return false;
+		}
+		if (targets.length > 1) move.spreadHit = true;
+
+		/** @type {((targets: Pokemon[], pokemon: Pokemon, move: ActiveMove) => (number | boolean | "" | undefined)[] | undefined)[]} */
+		let moveSteps = [
```

Comment:
I think I would prefix all 8 hit step functions with `hitStep`.

So `hitStepTryImmunity`, `hitStepTryHit`, etc.

Reply:
These function names are kind of vague without the namespacing.